pyproject.toml

```diff
@@ -85,6 +86,12 @@ docs = [
     "jupyterlite-pyodide-kernel",
     "nbformat",
 ]
+dev = [
```

Comment:
let's merge `test` into `dev` and move them (with docs) to the new "[dependency-groups]" (we don't need to build dev and docs from PyPI, just locally).
I don't think `isort` is useful as sorting is now done in `ruff`  
What's the reason to include `pydocstyle`?

Reply:
See my comment above on why not dependency-groups.

isort can go, did some experiments with pydocstyle, pushed by mistake